### PR TITLE
Add type annotations for public members with Scalafix

### PR DESCRIPTION
**Pull request by olafurpg**
Previously, Scalafix only checked for unused imports. Now, it also adds
type annotations for public members.

### CODE REVIEW

File: mtags/src/main/scala/scala/meta/internal/docstrings/ScaladocParser.scala [hunk 1]
```diff
@@ -1063,8 +1063,8 @@ object ScaladocParser {
 
     /* INLINES */
 
-    val OPEN_TAG = "^<([A-Za-z]+)( [^>]*)?(/?)>$".r
-    val CLOSE_TAG = "^</([A-Za-z]+)>$".r
+    val OPEN_TAG: Regex = "^<([A-Za-z]+)( [^>]*)?(/?)>$".r
+    val CLOSE_TAG: Regex = "^</([A-Za-z]+)>$".r
```
Comment (gabro) [inline mtags/src/main/scala/scala/meta/internal/docstrings/ScaladocParser.scala]: these are other examples where I don't see a great value.

Comment (olafurpg) [reply mtags/src/main/scala/scala/meta/internal/docstrings/ScaladocParser.scala]: Sure, but this particular is not a major regression either IMO. Once you get used to always having public types I personally find it quite helpful even in cases like this, now you don't even need to squint to observe the trailing `.r`

File: mtags/src/main/scala/scala/meta/internal/metals/Docstrings.scala [hunk 2]
```diff
@@ -19,7 +19,8 @@ import scala.meta.internal.mtags.GlobalSymbolIndex
  * Handles both javadoc and scaladoc.
  */
 class Docstrings(index: GlobalSymbolIndex) {
-  val cache = TrieMap.empty[String, SymbolDocumentation]
+  val cache: TrieMap[String, SymbolDocumentation] =
```
[the remark this answers is not in the document]
Comment (olafurpg) [reply mtags/src/main/scala/scala/meta/internal/metals/Docstrings.scala]: I agree this particular case is quite noisy, refactored this code to use `new TrieMap[K, V]()`, which the rewrite allows to be unannotated.

File: metals/src/main/scala/scala/meta/internal/metals/ClientCommands.scala [hunk 3]
```diff
@@ -7,7 +7,7 @@ import ch.epfl.scala.{bsp4j => b}
  */
 object ClientCommands {
 
-  val EchoCommand = Command(
+  val EchoCommand = new Command(
```
[the remark this answers is not in the document]
Comment (olafurpg) [reply metals/src/main/scala/scala/meta/internal/metals/ClientCommands.scala]: This was a manual fix, `new T` expressions always have type `T` so the rewrite leaves the unannotated by default. It's possible to let the rewrite also add annotations in this case but I don't think it improves readability much.

**Comment by olafurpg** [comment]
A agree, if it turns out to be more annoying than helpful then we can turn it off. My motivation for exploring this partly comes from wanting to see if we can somehow use the new `-Youtline` to improve 

* time to completions in new workspaces
* keeping the symbol table up to date for the presentation compiler during refactorings when the code doesn't compile and it off-sync with the classfiles on disk

**Comment by olafurpg** [comment]
If we're going to ask users to type annotate their public members then we should first try it out ourselves

**Comment by gabro** [comment]
Ah, that’s very interesting. Makes sense 👍

**Pull request merged**